src/index.js

```diff
@@ -102,6 +104,22 @@ export default async function loader(content, sourceMap, meta) {
       processOptions
     );
   } catch (error) {
+    // Check postcss versions to avoid using PostCSS 7.
+    if (
+      !hasExplicitDependencyOnPostCSS &&
+      postcssFactory().version.startsWith("7.")
+    ) {
+      // For caching reasons, we use the readFileSync function from the context, not the function from the `fs` module.
+      const pkg = readPackageJson(this.fs.readFileSync);
```

Comment:
You need to check that `pacjage.json` exists first (print nothing on no file).

Otherwise, it will throw Can’t read file error if package.json was in a different place.

Reply:
Thanks, done.